tests/lib/test-crio.bash.in

```diff
@@ -0,0 +1,214 @@
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# CC tests data
+TEST_DATA="${BATS_TEST_DIRNAME}/../../data/"
+
+# CRI-O URL and version
+CRIO_GITHUB="github.com/kubernetes-incubator/cri-o"
+CRIO_VERSION="3195f459"
+
+# CRI-O root directory
+OCID_ROOT="$GOPATH/src/$CRIO_GITHUB"
+
+# Path of the ocid binary.
+OCID_BINARY=${OCID_BINARY:-${OCID_ROOT}/ocid}
+# Path of the ocic binary.
+OCIC_BINARY=${OCIC_BINARY:-${OCID_ROOT}/ocic}
+# Path of the conmon binary.
+CONMON_BINARY=${CONMON_BINARY:-${OCID_ROOT}/conmon/conmon}
+# Path of the pause binary.
+PAUSE_BINARY=${PAUSE_BINARY:-${OCID_ROOT}/pause/pause}
+# Path of the default seccomp profile.
+SECCOMP_PROFILE=${SECCOMP_PROFILE:-${OCID_ROOT}/seccomp.json}
+# Name of the default apparmor profile.
+APPARMOR_PROFILE=${APPARMOR_PROFILE:-ocid-default}
+# Path of the bin2img binary.
+BIN2IMG_BINARY=${BIN2IMG_BINARY:-${OCID_ROOT}/test/bin2img/bin2img}
+# Path of the copyimg binary.
+COPYIMG_BINARY=${COPYIMG_BINARY:-${OCID_ROOT}/test/copyimg/copyimg}
+# Path of tests artifacts.
+ARTIFACTS_PATH="@CRIO_CACHE@"
+# Path of the checkseccomp binary.
+CHECKSECCOMP_BINARY=${CHECKSECCOMP_BINARY:-${OCID_ROOT}/test/checkseccomp/checkseccomp}
+
+# CRI-O Runtime
+RUNTIME=${RUNTIME:-cc-oci-runtime}
+RUNTIME_PATH=$(command -v $RUNTIME || true)
+RUNTIME_BINARY=${RUNTIME_PATH:-/usr/bin/cc-oci-runtime}
+
+TESTDIR=$(mktemp -d)
+if [ -e /usr/sbin/selinuxenabled ] && /usr/sbin/selinuxenabled; then
+    . /etc/selinux/config
+    filelabel=$(awk -F'"' '/^file.*=.*/ {print $2}' /etc/selinux/${SELINUXTYPE}/contexts/lxc_contexts)
+    chcon -R ${filelabel} $TESTDIR
+fi
+OCID_SOCKET="$TESTDIR/ocid.sock"
+OCID_CONFIG="$TESTDIR/ocid.conf"
+OCID_CNI_CONFIG="$TESTDIR/cni/net.d/"
+OCID_CNI_PLUGIN="/opt/cni/bin/"
+POD_CIDR="10.88.0.0/16"
+POD_CIDR_MASK="10.88.*.*"
+mkdir -p $OCID_CNI_CONFIG
+
+PATH=$PATH:$TESTDIR
+
+# Make sure we have a copy of the CRI-O code.
+if ! [ -d "$OCID_ROOT" ]; then
+    #    mkdir -p "$OCID_ROOT"
+    echo "Cloning CRI-O repo..."
+    go get $CRIO_GITHUB || true
+    pushd $OCID_ROOT && git reset --hard $CRIO_VERSION && popd
+fi
+
+# Build the CRI-O binaries
+echo "Building CRI-O...($GOPATH)"
+make -C ${OCID_ROOT} binaries
+
+cp "$CONMON_BINARY" "$TESTDIR/conmon"
+
+# Make sure we have a copy of the redis:latest image.
+if ! [ -d "$ARTIFACTS_PATH"/redis-image ]; then
+    mkdir -p "$ARTIFACTS_PATH"/redis-image
+    echo "Copying redis image (Only needed during the first run)..."
+    if ! "$COPYIMG_BINARY" --import-from=docker://redis --export-to=dir:"$ARTIFACTS_PATH"/redis-image --signature-policy="$TEST_DATA"/policy.json ; then
+        echo "Error pulling docker://redis"
+        rm -fr "$ARTIFACTS_PATH"/redis-image
+        exit 1
+    fi
+fi
+
+# Retry a command $1 times until it succeeds. Wait $2 seconds between retries.
+function retry() {
+	local attempts=$1
+	shift
+	local delay=$1
+	shift
+	local i
+
+	for ((i=0; i < attempts; i++)); do
+		run "$@"
```

Comment:
This is technically a bats script, not a bash script, as it's calling `run`, which is only available under bats. I'd either add a comment to the file to state that, or rename it to `.bats.in`.